pkg/gadgets/trace/exec/tracer/tracer_test.go

```diff
@@ -0,0 +1,219 @@
+		t.Fatal("returned tracer was nil")
+	}
+	defer tracer.Stop()
+
+	// check that a double stop doesn't cause issues
+	tracer.Stop()
+	tracer.Stop()
+}
+
+func TestTracer(t *testing.T) {
+	if unix.Getuid() != 0 {
+		t.Skip("test requires root")
+	}
+
+	// variable to save the value of the created process PID
+	var catPid int
+
+	manyArgs := []string{}
+	// 19 is DEFAULT_MAXARGS - 1
+	for i := 0; i < 19; i++ {
+		manyArgs = append(manyArgs, "/dev/null")
+	}
+
+	// function to generate an event used most of the times
+	generateEvent := func() error {
+		cmd := exec.Command("/bin/cat", "/dev/null")
+		if err := cmd.Run(); err != nil {
+			return err
+		}
+
+		catPid = cmd.Process.Pid
+
+		return nil
+	}
+
+	testRunner := utilstest.NewTestRunner[types.Event](t)
+
+	tests := map[string]utilstest.Test[types.Event]{
+		"test without filtering": {
+			GetTracer: func(runner *utilstest.Runner, eventCallback func(event types.Event)) utilstest.Tracer {
+				tracer, err := NewTracer(&Config{}, nil, eventCallback)
+				if err != nil {
+					testRunner.Fatalf("error creating tracer: %s", err)
+				}
+				return tracer
+			},
+			GenerateEvent: generateEvent,
+			GetAnyExpectedEvent: func(runner *utilstest.Runner) *types.Event {
+				return &types.Event{
+					Event: eventtypes.Event{
+						Type: eventtypes.NORMAL,
+					},
+					Pid:       uint32(catPid),
+					Ppid:      uint32(runner.Info.Pid),
+					UID:       uint32(runner.Info.UID),
+					MountNsID: runner.Info.MountNsID,
+					Retval:    0,
+					Comm:      "cat",
+					Args:      []string{"/bin/cat", "/dev/null"},
+				}
+			},
+		},
+		"test filtering (no event should be captured)": {
+			GetTracer: func(runner *utilstest.Runner, eventCallback func(event types.Event)) utilstest.Tracer {
+				m, err := utilstest.CreateMntNsSetMap(0)
+				if err != nil {
+					testRunner.Fatalf("error creating ebpf map: %s", err)
+				}
+
+				tracer, err := NewTracer(&Config{MountnsMap: m}, nil, eventCallback)
+				if err != nil {
+					testRunner.Fatalf("error creating tracer: %s", err)
+				}
+				return tracer
+			},
+			GenerateEvent: generateEvent,
+		},
+		"test filtering": {
+			GetTracer: func(runner *utilstest.Runner, eventCallback func(event types.Event)) utilstest.Tracer {
+				m, err := utilstest.CreateMntNsSetMap(runner.Info.MountNsID)
+				if err != nil {
+					testRunner.Fatalf("error creating ebpf map: %s", err)
+				}
+
+				tracer, err := NewTracer(&Config{MountnsMap: m}, nil, eventCallback)
+				if err != nil {
+					testRunner.Fatalf("error creating tracer: %s", err)
+				}
+				return tracer
+			},
+			GenerateEvent: generateEvent,
+			GetExpectedEvent: func(runner *utilstest.Runner) *types.Event {
+				return &types.Event{
+					Event: eventtypes.Event{
+						Type: eventtypes.NORMAL,
+					},
+					Pid:       uint32(catPid),
+					Ppid:      uint32(runner.Info.Pid),
+					UID:       uint32(runner.Info.UID),
+					MountNsID: runner.Info.MountNsID,
+					Retval:    0,
+					Comm:      "cat",
```

Comment:
`runner.Info.Comm` wouldn't be the same?

Reply:
`runner.Info.Comm` is the name of the process executing the events (parent process). `Comm` is the name of the executed process.